### PR TITLE
feat: allow pydantic v2 usage

### PR DESCRIPTION
**Pull request by cfbevan**
* feat: start migration to pydantic v2
* update tests for pydantic v2

Followed https://docs.pydantic.dev/latest/migration/#using-pydantic-v1-features-in-a-v1v2-environment

Note that current documentation references a CONTRIBUTING.md that does not exist. The GitHub Pages site encourages reaching out for instructions on contributing, but no contact information is listed. Please feel free to comment on any changes needed.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -100,7 +100,7 @@ version = "0.2.1"
 [tool.poetry.dependencies]
 aioarango = "^1.0.0"
 indexed = "^1.3.0"
-pydantic = "==1.10.12"
+pydantic = ">=1.10.17"
```
Comment (nadobando) [inline pyproject.toml:103]: maybe we can already set this to 
```suggestion
pydantic = "<2"
```

**Review by nadobando — APPROVED** [review]
gr8 initiative @cfbevan

**Comment by cfbevan** [comment]
I'm a bit confused by the CI checks. They ran locally fine. I will take a look at the errors and update as needed.

File: pyproject.toml (hunk 1 above)
Comment (cfbevan) [reply pyproject.toml:103]: The documentation claims that the `v1` modules were not added till 1.10.17. Another option is to update the imports to something like this:

```python
try:
    from pydantic.v1.fields import ModelField
except ImportError:
    from pydantic.fields import ModelField 
```

**Comment by cfbevan** [comment]
I see my issue. The CI/CD uses 3.9 and I was using 3.11 locally.

**Comment by cfbevan** [comment]
All tests passed. There is something wrong with the tokens.

**Comment by nadobando** [comment]
`CI / Tests` is failing because of the token permissions, I'll will merge and open an issue :) 
thanks for you sharing

**Pull request merged**